projects/igniteui-angular/src/lib/select/select.component.ts

```diff
@@ -312,9 +377,51 @@ export class IgxSelectComponent extends IgxDropDownComponent implements IgxSelec
 
     /** @hidden @internal */
     public onBlur(): void {
+        if (this.ngControl && !this.ngControl.valid) {
```

Comment:
Could `onBlur` call `this.onStatusChanged()`? The logic seems to be the same

Reply:
Actually I prefer to separate the initial+update validation logic from the one updating the states onBlur

Reply:
Makes sense, I missed the `STATE.INITIAL` part - it'll behave like our components